dora/core/client/fs/src/main/java/alluxio/client/file/PositionReadFileInStream.java

```diff
@@ -153,13 +170,19 @@ private CallTrace(long pos, int length) {
 
   /**
    * Constructor.
-   * @param reader
-   * @param length
+   * @param reader the position reader
+   * @param uriStatus the uri status
+   * @param client the dora client
    */
-  public PositionReadFileInStream(PositionReader reader,
-      long length) {
+  public PositionReadFileInStream(
+      PositionReader reader,
+      URIStatus uriStatus,
+      DoraCacheClient client
+  ) {
+    mClient = client;
+    mURIStatus = uriStatus;
```

Comment:
the `mURIStatus` is cached so `uriStatus.getInAlluxioPercentage()` won't be updated throughout the lifetime of the stream. Therefore, this check can be merged into a single variable `mPreloadEnabled`:

```java
mPreloadEnabled = Configuration.getBoolean(PropertyKey.USER_POSITION_READER_PRELOAD_DATA_ENABLED)
&& uriStatus.getLength() > mDataPreloadFileSizeThreshold
&& uriStatus.getInAlluxioPercentage() != 100;

```